### PR TITLE
Add versions allowUnsafeNewFunction, allowUnsafeNewFunction for use with async flows

### PR DESCRIPTION
**Pull request by lucasmarshall**

### CODE REVIEW

**Comment by lucasmarshall** [comment]
@kevinsawicki @nathansobo if you could take a look at this, that would be great!

**Comment by nathansobo** [comment]
Could you add some basic specs to this just to make sure these functions basically work? Thanks.

**Comment by nathansobo** [comment]
They should also be added to the README.

File: src/loophole.coffee [hunk 1]
```diff
@@ -28,3 +28,21 @@ exports.Function = (paramLists..., body) ->
...
+    callback = -> global.eval = previousEval
+    fn(callback)
+  catch e
+    global.eval = previousEval
+
+exports.allowUnsafeNewFunctionAsync = (fn) ->
+  previousFunction = global.Function
+  try
+    global.Function = exports.Function
+    callback = -> global.eval = global.eval = previousFunction
```
Comment (kevinsawicki) [inline src/loophole.coffee]: Is there a reason that `global.eval` is assigned twice on this line?

**Comment by stevemao** [comment]
I need this in my package so it would be great if this can be merged asap @lucasmarshall

**Comment by stevemao** [comment]
The implementation could be simplified as `allowUnsafeEval` could depend on `allowUnsafeEvalAsync`.
Or we could sniff the arity on `fn`. if there is no `callback` its sync. async otherwise.

**Comment by nathansobo** [comment]
@stevemao If you want to push it forward with some spec coverage and your suggested improvement I'd be happy to work with you on merging this.

**Comment by stevemao** [comment]
@nathansobo 
I have sent a PR #7

**Pull request closed without merging**